File: backend/internal/service/clinical_note_service.go
package service

import (
	"time"

	"github.com/ZarishSphere-Platform/zarish-his/internal/models"
	"github.com/ZarishSphere-Platform/zarish-his/internal/repository"
)

type ClinicalNoteService struct {
	repo *repository.ClinicalNoteRepository
}

func NewClinicalNoteService(repo *repository.ClinicalNoteRepository) *ClinicalNoteService {
	return &ClinicalNoteService{repo: repo}
}

func (s *ClinicalNoteService) CreateNote(note *models.ClinicalNote) (*models.ClinicalNote, error) {
	if note.NoteDate.IsZero() {
		note.NoteDate = time.Now()
	}
	if note.Status == "" {
		note.Status = "draft"
	}
	return s.repo.Create(note)
}

func (s *ClinicalNoteService) GetNoteByID(id uint) (*models.ClinicalNote, error) {
	return s.repo.FindByID(id)
}

func (s *ClinicalNoteService) UpdateNote(note *models.ClinicalNote) (*models.ClinicalNote, error) {
	return s.repo.Update(note)
}

func (s *ClinicalNoteService) SignNote(id uint, userID uint) (*models.ClinicalNote, error) {
	note, err := s.repo.FindByID(id)
	if err != nil {
		return nil, err
	}
	note.Sign(userID)
	return s.repo.Update(note)
}

func (s *ClinicalNoteService) ListEncounterNotes(encounterID uint) ([]*models.ClinicalNote, error) {
	return s.repo.ListByEncounter(encounterID)
}

func (s *ClinicalNoteService) ListPatientNotes(patientID uint, limit int) ([]*models.ClinicalNote, error) {
	return s.repo.ListByPatient(patientID, limit)
}

File: frontend/src/services/clinicalNoteService.ts
import api from './api';
import type { ClinicalNote } from '../types';

export const ClinicalNoteService = {
  create: async (note: Partial<ClinicalNote>): Promise<ClinicalNote> => {
    const response = await api.post<ClinicalNote>('/clinical-notes', note);
    return response.data;
  },

  getById: async (id: number): Promise<ClinicalNote> => {
    const response = await api.get<ClinicalNote>(`/clinical-notes/${id}`);
    return response.data;
  },

  update: async (id: number, note: Partial<ClinicalNote>): Promise<ClinicalNote> => {
    const response = await api.put<ClinicalNote>(`/clinical-notes/${id}`, note);
    return response.data;
  },

  sign: async (id: number, userId: number): Promise<ClinicalNote> => {
    const response = await api.post<ClinicalNote>(`/clinical-notes/${id}/sign`, { user_id: userId });
    return response.data;
  },

  listByEncounter: async (encounterId: number): Promise<ClinicalNote[]> => {
    const response = await api.get<ClinicalNote[]>(`/encounters/${encounterId}/clinical-notes`);
    return response.data;
  },

  listByPatient: async (patientId: number, limit = 20): Promise<ClinicalNote[]> => {
    const response = await api.get<ClinicalNote[]>(`/patients/${patientId}/clinical-notes`, { params: { limit } });
    return response.data;
  },
};
